src/metabase/api/dataset.clj

```diff
@@ -83,25 +84,37 @@
      (api/defendpoint POST [\"/:export-format\", :export-format export-format-regex]"
   (re-pattern (str "(" (str/join "|" (map u/qualified-name (qp.streaming/export-formats))) ")")))
 
+(def ^:private column-ref-regex #"\[\"ref\",\[(?:\"field\",\d+,null|\"expression\",\".+\")\]\]")
+
+(defn- parse-viz-setting-key
```

Comment:
It doesn't seem like this function actually parses anything??

Reply:
Yeah I was thinking of it as "the key function for parsing viz settings" but I can rename it